includes/class-wc-payments-webhook-processing-service.php

```diff
@@ -304,34 +283,33 @@ private function process_webhook_refund_updated( $event_body ) {
 			foreach ( $wc_refunds as $wc_refund ) {
 				$wcpay_refund_id = $this->order_service->get_wcpay_refund_id_for_order( $wc_refund );
 				if ( $refund_id === $wcpay_refund_id ) {
-					// Delete WC Refund.
-					$wc_refund->delete();
+					$matched_wc_refund = $wc_refund;
 					break;
 				}
 			}
 		}
 
-		// Update order status if order is fully refunded.
-		$current_order_status = $order->get_status();
-		if ( Order_Status::REFUNDED === $current_order_status ) {
-			$order->update_status( Order_Status::FAILED );
-		}
-
-		$order->add_order_note( $note );
-		$this->order_service->set_wcpay_refund_status_for_order( $order, 'failed' );
-		$order->save();
-
-		try {
-			$failure_reason = $this->read_webhook_property( $event_object, 'failure_reason' );
-
-			if ( 'insufficient_funds' === $failure_reason ) {
-				$this->order_service->handle_insufficient_balance_for_refund(
-					$order,
-					$amount
-				);
-			}
-		} catch ( Exception $e ) {
-			Logger::debug( 'Failed to handle insufficient balance for refund: ' . $e->getMessage() );
+		// Refund update webhook events can be either failed, cancelled (basically it's also a failure but triggered by the merchant), succeeded only.
+		switch ( $status ) {
+			case Refund_Status::FAILED:
+				$this->order_service->handle_failed_refund( $order, $refund_id, $amount, $currency, $matched_wc_refund );
+				if (
```

Comment:
I think there's an opportunity to handle more `failure_reason` codes as listed in the Stripe [docs](https://docs.stripe.com/api/refunds/object#refund_object-failure_reason) (i.e. `expired_or_canceled_card`). IMO it will be useful for the merchant. But I think we can tackle it on a follow up issue, since we will probably need to refactor how we are handling the failed refund failure codes to use a holistic approach. What do you think? 
Note: the failure reason `expired_or_canceled_card` is already visible in the timeline view

Reply:
I opened https://github.com/Automattic/woocommerce-payments/issues/10645 to tackle it

Reply:
Thanks @mgascam for creating the issue.